packages/expo-modules-core/android/src/main/cpp/CachedReferencesRegistry.h

```diff
@@ -77,4 +75,52 @@ class CachedReferencesRegistry {
     const std::vector<std::pair<std::string, std::string>> &methods
   );
 };
+
+/**
+ * Registry used to store references to often used JS objects like Promise.
+ * The object lifetime should be bound with the JS runtime.
+ */
+class JSCachedReferencesRegistry {
```

Comment:
`JSReferencesCache`
Can we make it common for both platforms? For now maybe just move it to separate file